File: frontend/src/app/utils/course-status.ts
/**
 * Status przebiegu kursu
 */
export enum CourseStatus {

  MEDICAL_EXAMS,
  DOCUMENTS_SUBMISSION,
  LECTURES,
  THEORY_INTERNAL_EXAM,
  DRIVING_LESSONS,
  PRACTICAL_INTERNAL_EXAM,
  STATE_EXAMS,
  FINISHED

}

export namespace CourseStatus {

  export function translate(status: CourseStatus): string {
    let translator: string;
    switch (CourseStatus[status] as unknown) {
      case CourseStatus.MEDICAL_EXAMS:
        translator = "Badania lekarskie";
        break;
      case CourseStatus.DOCUMENTS_SUBMISSION:
        translator = "Złożenie dokumentów";
        break;
      case CourseStatus.LECTURES:
        translator = "Zajęcia teoretyczne (wykłady)";
        break;
      case CourseStatus.THEORY_INTERNAL_EXAM:
        translator = "Teoretyczny egzamin wewnętrzny";
        break;
      case CourseStatus.DRIVING_LESSONS:
        translator = "Zajęcia praktyczne (jazdy szkoleniowe)";
        break;
      case CourseStatus.PRACTICAL_INTERNAL_EXAM:
        translator = "Praktyczny egzamin wewnętrzny";
        break;
      case CourseStatus.STATE_EXAMS:
        translator = "Egzaminy państwowe";
        break;
      case CourseStatus.FINISHED:
        translator = "Odebranie prawa jazdy i zakończenie kursu";
        break;
      default:
        translator = "";
        break;
    }
    return translator;
  }

}

File: src/main/java/com/project/webapp/drivingschool/data/utils/CourseStatus.java
package com.project.webapp.drivingschool.data.utils;

/**
 * Status przebiegu kursu nauki jazdy
 */
public enum CourseStatus {

    /**
     * Przeprowadzenie badań lekarskich
     */
    MEDICAL_EXAMS,

    /**
     * Złożenie dokumentu PKK i pisemnej zgody rodziców (opcjonalnie)
     */
    DOCUMENTS_SUBMISSION,

    /**
     * Zajęcia teoretyczne (wykłady)
     */
    LECTURES,

    /**
     * Teoretyczny egzamin wewnętrzny
     */
    THEORY_INTERNAL_EXAM,

    /**
     * Zajęcia praktyczne (jazdy szkoleniowe)
     */
    DRIVING_LESSONS,

    /**
     * Praktyczny egzamin wewnętrzny
     */
    PRACTICAL_INTERNAL_EXAM,

    /**
     * Egzaminy państwowe
     */
    STATE_EXAMS,

    /**
     * Odebranie prawa jazdy i zakończenie kursu
     */
    FINISHED

}
